### ISSUE TITLE
Remove Cross-Namespace OwnerReferences & Ensure Resource Deletion

### ISSUE THREAD

**Issue by travis-minke-sap** [kyma-incubator/knative-kafka#72]
The current implementation is creating the runtime resources (Channel/Dispatcher Service/Deployment) in the **knative-eventing** namespace instead of in the "user" namespace where the KafkaChannel and Knative Subscription exist.  

K8S documentation (below) states that this is not allowed and OwnerReferences should not span namespaces.  Therefore we'd like to remove these controller references and still have guaranteed resource cleanup. Now that we're using the Knative-Eventing controller framework we should be able to achieve this with the available informers and such.

```
Note: Cross-namespace owner references are disallowed by design. This means: 
1) Namespace-scoped dependents can only specify owners in the same namespace, 
and owners that are cluster-scoped. 2) Cluster-scoped dependents can only specify 
cluster-scoped owners, but not namespace-scoped owners.
```

**Note:** It is believed knative-eventing is doing similar things in various implementation (broker/trigger/etc) and we should probably use that as an example.

**Note:** This should focus on the Dispatcher Service/Deployment since the pending refactor of Channel Service/Deployment reconciliation into a separate controller (based on the Kafka Secret) will inherently NOT be cross-namespace and can rely on normal K8S OwnerReferences & Garbage Collection. See Issue #73

**Comment by travis-minke-sap** [comment]
Note - This will likely be resolved by the work done in #73 by labelling our K8S resources and making use of the knative.dev controller.EnqueueLabelOfNamespaceScopedResource() functionality.  This does not REMOVE the cross-reference namespaces as indicated in the story description but is probably adequate ?

**Comment by travis-minke-sap** [comment]
closing - as described above resolved by delivery for #73 - cross-namespace issues resolved.
